src/foundry/client/apps/forms/journal-page-sheet.d.ts

```diff
@@ -0,0 +1,292 @@
+    protected _renderInner(data: object): Promise<JQuery<HTMLElement>>;
+    _getSecretContent(secret: HTMLElement): string;
+    // FIXME: revisit this when it's added to its ancestor classes
+    _updateSecret(secret: HTMLElement, content: string): Promise<typeof this>;
+    // FIXME: this should be async once the parent classes are updated
+    protected _updateObject(event: Event, formData: object): Promise<unknown>;
+    /**
+     * Lazily convert text formats if we detect the document being saved in a different format.
+     * @param renderData - Render data.
+     */
+    protected _convertFormats(renderData: ReturnType<typeof this["getData"]>): void;
+
+    /* -------------------------------------------- */
+    /*  HTML Editing                                */
+    /* -------------------------------------------- */
+
+    /**
+     * Update the UI appropriately when receiving new steps from another client.
+     */
+    onNewSteps(): void;
+  }
+  namespace JournalTextPageSheet {
+    interface Options extends JournalPageSheet.Options {
+      editor: {
+        engine: Required<TextEditor.Options>["engine"];
+        collaborate: boolean;
+        content: ReturnType<typeof TextEditor.enrichHTML>;
+      };
+    }
+  }
+
+  /**
+   * The Application responsible for displaying and editing a single {@link JournalEntryPage} image document.
+   */
+  class JournalImagePageSheet extends JournalPageSheet<JournalImagePageSheet.Options> {
+    /**
+     * @defaultValue
+     * ```typescript
+     * const options = super.defaultOptions;
+     * options.classes.push("image");
+     * options.height = "auto";
+     * ```
+     */
+    static override get defaultOptions(): JournalImagePageSheet.Options;
+  }
+  namespace JournalImagePageSheet {
+    type Options = JournalPageSheet.Options;
+  }
+  /**
+   * The Application responsible for displaying and editing a single {@link JournalEntryPage} video document.
+   */
+  class JournalVideoPageSheet extends JournalPageSheet<JournalVideoPageSheet.Options> {
+    activateListeners(html: JQuery<HTMLElement>): void;
+    /**
+     * Get the YouTube player parameters depending on whether the sheet is being viewed or edited.
+     */
+    protected _getYouTubeVars(): JournalVideoPageSheet.YouTubeVars;
+    protected _getSubmitData(updateData?: object | null | undefined): Record<string, unknown>;
+    /**
+     * Convert time components to a timestamp in seconds.
+     * @param param0 - The time components.
+     * @returns The timestamp, in seconds.
+     */
+    protected _timeComponentsToTimestamp({ h, m, s }: Partial<JournalVideoPageSheet.TimeComponents>): number;
+    /**
+     * Convert a timestamp in seconds into separate time components.
+     * @param timestamp - The timestamp, in seconds.
+     * @returns The individual time components
+     */
+    protected _timestampToTimeComponents(timestamp: number): JournalVideoPageSheet.TimeComponents;
+  }
+  namespace JournalVideoPageSheet {
```

Comment:
nit; same general idea here: between classes, functions, really just blocks in general there should be an empty newline.